### PR TITLE
Install Mac OS X dependencies

### PR DESCRIPTION
**Pull request by beholders-eye**
Ressurected InstallPlatformDependencies, as a kind of a generic function

### CODE REVIEW

**Comment by beholders-eye** [comment]
I'll be honest. Never thought of that. But I think it goes in the same lines of the question about `GOPRIVATE` (and the respective answer). To run Python you'll need that, we're just making it a bit easier, IMO. To run Ruby/Homebrew too. That piece of code was already in YB CLI since the first version.

WDYT @zombiezen ?

**Comment by beholders-eye** [comment]
There's no best answer to this, but this should fix Python installations in Mac OS X. If you like, I can test it better

**Comment by zombiezen** [comment]
I'm with @jamesnaftel on this one. If there are system requirements we need to ensure before using the tool, we should be providing instructions to users or, even better, provide macOS binary packages or `brew` packaging.

**Comment by beholders-eye** [comment]
> I'm with @jamesnaftel on this one. If there are system requirements we need to ensure before using the tool, we should be providing instructions to users or, even better, provide macOS binary packages or `brew` packaging.

For the time being, I'll add instructions to README.md.

@zombiezen what do you mean by macOS binary packages. `yb` is released as Mac OS binary too. Do you meant a `.dmg` or `.pkg`? Did you suggest that we add `/Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg` to something like a `yb-cli.pkg`?

**Comment by zombiezen** [comment]
It's been a long time since I've done macOS packaging, but IIRC the `.pkg` format permits distributing other `.pkg` files as dependencies and this behaves more nicely with macOS sysadmin things.

File: README.md [hunk 1]
```diff
@@ -161,6 +161,11 @@ You can change yb to other channels by using `-channel`:
 
 Currently, the _inner_ yb that runs inside the build container will use the stable channel by default. To change that, set the environment variable `YB_UPDATE_CHANNEL=unstable` before running yb.
 
+# Mac OS X dependency
```
Comment (zombiezen) [inline README.md:164]: Let's add this to the "How to use it" section so the user isn't surprised later.

File: README.md [hunk 2]
```diff
@@ -161,6 +161,11 @@ You can change yb to other channels by using `-channel`:
...
+
+Some build packs (Python, Ruby and Homebrew) needs Mac OS SDK Headers to be installed:
```
Comment (zombiezen) [inline README.md:166]: Correct styling is "macOS".

```suggestion
Some build packs (Python, Ruby and Homebrew) needs macOS SDK Headers to be installed:
```

**Pull request merged**